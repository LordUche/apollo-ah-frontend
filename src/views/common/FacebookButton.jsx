import React from 'react';
import FacebookLogin from 'react-facebook-login/dist/facebook-login-render-props';
import PropTypes from 'prop-types';
import facebook from '../../images/icons/facebook.svg';
import ImageButton from './ImageButton';

/**
 * A function representing FacebookButton
 *
 * @param {{Function}} handleResponse - A response handler callback
 * @returns {JSX.Element} - DOM element
 */
const FacebookButton = ({ handleResponse }) => (
  <FacebookLogin
    appId={process.env.FACEBOOK_APP_ID}
    autoLoad={false}
    fields="name,email,picture"
    callback={handleResponse}
    render={renderProps => (
      <ImageButton
        imgSrc={facebook}
        onClick={renderProps.onClick}
        altText="login with facebook"
      />
    )}
  />
);

FacebookButton.propTypes = {
  handleResponse: PropTypes.func.isRequired,
};

export default FacebookButton;
