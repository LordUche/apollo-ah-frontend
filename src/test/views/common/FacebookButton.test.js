/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FacebookButton from '../../../views/common/FacebookButton';

describe('<FacebookButton />', () => {
  it('should render facebook button', () => {
    const enzymeWrapper = shallow(<FacebookButton handleResponse={jest.fn()} />);
    expect(enzymeWrapper.exists()).toBe(true);
    expect(toJson(enzymeWrapper)).toMatchSnapshot();
  });
});
