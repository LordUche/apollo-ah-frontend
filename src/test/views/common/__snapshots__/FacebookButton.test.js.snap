// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FacebookButton /> should render facebook button 1`] = `
<t
  authType=""
  autoLoad={false}
  callback={[MockFunction]}
  cookie={false}
  disableMobileRedirect={false}
  fields="name,email,picture"
  isMobile={false}
  language="en_US"
  onFailure={null}
  redirectUri="http://localhost/"
  render={[Function]}
  responseType="code"
  returnScopes={false}
  scope="public_profile,email"
  state="facebookdirect"
  version="2.3"
  xfbml={false}
/>
`;
